src/streaming/controllers/ProtectionController.js

```diff
@@ -418,8 +421,62 @@ MediaPlayer.dependencies.ProtectionController = function () {
                 data:event.data,
                 error:null
             });
+        },
+        
+        // Resolve all init data, for Adobe Access the init may need to be downloaded from
+        // a given URL in the mpd.
+        getPsshForSupportedKeySystems = function(supportedKS) {
+
+            var self = this,
+                supportedKSInit = {};
+            supportedKSInit[MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE] = function(event) {
+                var ksIdx = 0,
+                    pending = false;
+                while( ksIdx < supportedKS.length) {
+                    if (event.data.requestData.schemeIdUri.toLowerCase() === supportedKS[ksIdx].ks.schemeIdURI) {
+                        if (!event.error) {
+                            supportedKS[ksIdx].initData = event.data.message;
+                        }
+                        else {
+                            self.log("DRM: Could not download PSSH data for keysystem : " + supportedKS[ksIdx].ks.schemeIdURI);
+                            supportedKS.splice(ksIdx);
+                            continue;
+                        }
+                        supportedKS[ksIdx].ks.unsubscribe(MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE, supportedKSInit);
+                    }
+                    else if (supportedKS[ksIdx].initData === "pending") {
+                        pending = true;
+                    }
+                    
+                    ksIdx++;
+                }
+                
+                if (!pending) {
+                    pendingPsshData = false;
+
+                    selectKeySystem.call(self, supportedKS, true);
+                }
+                
+            };
+            
+            if (supportedKS && supportedKS.length > 0) {
+                var ksIdx = 0;
+                
+                for(ksIdx = 0; ksIdx < supportedKS.length; ++ksIdx) {
+                    if (supportedKS[ksIdx].initData === "pending") {
+                        supportedKS[ksIdx].ks.subscribe(MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE, supportedKSInit);
+                        pendingPsshData = true;
+                    }
+                }
+            }
+            
+            if (!pendingPsshData) {
+                selectKeySystem.call(self, supportedKS, true);
+            }
+
         };
```

Comment:
@rashmimittal We'll have to come back and revisit this one after the ES6 refactor work is merged into dev.  Since the entire event system will be changed.

Also, the latest EME spec is entirely dependent on Promises.  So, I really hope we can start using them more throughout the dash.js code for simple async operations.